core.js

```diff
@@ -105,69 +68,90 @@ function applySubraceBonuses(char){
-            heightAndWeight[1] = additionalHeightAndWeight[1] + 110 + " lbs";
-            break;
-        case race === "tiefling":
-            additionalHeightAndWeight = returnHeightAndWeight("2d8", "2d4");
-            heightAndWeight[0] = additionalHeightAndWeight[0] + 4.9 + " ft";
-            heightAndWeight[1] = additionalHeightAndWeight[1] + 110 + " lbs";
-            break;
-
-        default: return "invalid race provided as function arguement [returnCharacterHeight()]";
-
-
-    }
-
-    return heightAndWeight;
-
+/**-------------------------------------------------------------------------------
+ * @description Calculates character `height`. The calculation adds a 'base height' 
+ * with a dice throw specific to each race. The base values are retrieve from:
+ *  https://www.reddit.com/r/DnD/comments/abebq0/chart_with_heightweight_ranges_and_averages_for/ 
+ * @param {String} race - the character race.
+ * @returns {Number} the calculated height in ft.  
+*///------------------------------------------------------------------------------
+function calculateCharacterHeight(race) {
+  let finalHeight = 0.0;
+  switch (race) {
+      case "dragonborn":
+          finalHeight = 5.6 + parseFloat((utils.diceThrow(2,8)/ 12).toFixed(1));
+      break;
+      case "dwarf":
+          finalHeight = 3.8 + parseFloat((utils.diceThrow(2,8)/ 12).toFixed(1));
+      break;
+      case "elf":
+          finalHeight = 4.6 + parseFloat((utils.diceThrow(2,10)/ 12).toFixed(1));
+      break;
+      case "gnome":
+          finalHeight = 2.11 + parseFloat((utils.diceThrow(2,4)/ 12).toFixed(1));
+      break;
+      case "half-elf":
+          finalHeight = 4.9 + parseFloat((utils.diceThrow(2,8)/ 12).toFixed(1));
+      break;
+      case "half-orc":
+          finalHeight = 4.10 + parseFloat((utils.diceThrow(2,10)/ 12).toFixed(1));
+      break;
+      case "halfling":
+          finalHeight = 2.7 + parseFloat((utils.diceThrow(2,4)/ 12).toFixed(1));
+      break;
+      case "human":
+          finalHeight = 4.8 + parseFloat((utils.diceThrow(2,10)/ 12).toFixed(1));
+      break;
+      case "tiefling":
+          finalHeight = 4.9 + parseFloat((utils.diceThrow(2,8)/ 12).toFixed(1));
+      break;
+      default: return "invalid race provided as function arguement [returnCharacterHeight()]";
+  }
+  return finalHeight;
+};
+
+/**-------------------------------------------------------------------------------
+ * @description Calculates character `weight`. The calculation adds a 'base height' 
+ * with a dice throw specific to each race. The base values are retrieve from:
+ *  https://www.reddit.com/r/DnD/comments/abebq0/chart_with_heightweight_ranges_and_averages_for/ 
+ * @param {String} race - the character race.
+ * @returns {Number} the calculated weight in lbs.  
+*///------------------------------------------------------------------------------
+function calculateCharacterWeight(race) {
```

Comment:
So the table you used as a source didn't explain how the weight figure is calculated, and my previous code was also incorrect so I wasn't helping you out there either!

While a player can just roll or pick a random number, the method described in the players hand book to calculate height and weight is as follows:

First step is to calculate the players height, using a hill dwarf as an example:

Base Height: 3'8"
Height Modifier: 2d4

Base Weight: 115lb
Weight Modifier: (result of Height Modifier) x 2d6

So if I rolled a height modifier of 6 I would calculate the players weight as 6 x 2d6

This would also explain the weight ranges described in your source table, rolling minimum values on both height modifier and weight modifier would result in : 115 + (2x2) = 119

And max values would be 115 + (8x12) = 211

hopefully that makes sense 👍

Reply:
Oh, great I will fix it and then send a new commit to this PR.